com.avaloq.tools.ddk.xtext.builder/src/com/avaloq/tools/ddk/xtext/builder/MonitoredClusteringBuilderState.java

```diff
@@ -362,28 +362,27 @@ protected void propagateDependencyChains(final Set<URI> toBeUpdated, final IReso
     Collection<IResourceDescription.Delta> deltas = doClean(toBeRemovedCopy, subMonitor.newChild(1));
 
     final ResourceDescriptionsData newData = getCopiedResourceDescriptionsData();
-    ResourceDescriptionChangeEvent event = null;
+
+    checkForCancellation(monitor);
     try {
-      checkForCancellation(monitor);
       for (IResourceDescription.Delta delta : deltas) {
         newData.removeDescription(delta.getOld().getURI());
       }
-      event = new ResourceDescriptionChangeEvent(deltas);
+      ResourceDescriptionChangeEvent event = new ResourceDescriptionChangeEvent(deltas);
       checkForCancellation(monitor);
       updateDeltas(event.getDeltas(), null, subMonitor.newChild(1));
       // update the reference
       setResourceDescriptionsData(newData, monitor);
       // CHECKSTYLE:CHECK-OFF IllegalCatch
+      notifyListeners(event);
+      return event.getDeltas();

```

Comment:
not related to your change, but can you pls fix the typo on line 380? (CHEKC)

Reply:
done